ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/DefaultAuthenticationService.java

```diff
@@ -0,0 +1,294 @@
+import org.hyperledger.besu.ethereum.api.jsonrpc.websocket.WebSocketConfiguration;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.Nullable;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.JWTOptions;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.auth.authentication.AuthenticationProvider;
+import io.vertx.ext.auth.jwt.JWTAuth;
+import io.vertx.ext.auth.jwt.JWTAuthOptions;
+import io.vertx.ext.web.RoutingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Provides authentication handlers for use in the http and websocket services */
+public class DefaultAuthenticationService implements AuthenticationService {
+
+  public static final String USERNAME = "username";
+  private final JWTAuth jwtAuthProvider;
+  @VisibleForTesting public final JWTAuthOptions jwtAuthOptions;
+  private final Optional<AuthenticationProvider> credentialAuthProvider;
+  private static final JWTAuthOptionsFactory jwtAuthOptionsFactory = new JWTAuthOptionsFactory();
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultAuthenticationService.class);
+
+  public DefaultAuthenticationService(
+      final JWTAuth jwtAuthProvider,
+      final JWTAuthOptions jwtAuthOptions,
+      final Optional<AuthenticationProvider> credentialAuthProvider) {
+    this.jwtAuthProvider = jwtAuthProvider;
+    this.jwtAuthOptions = jwtAuthOptions;
+    this.credentialAuthProvider = credentialAuthProvider;
+  }
+
+  /**
+   * Creates a ready for use set of authentication providers if authentication is configured to be
+   * on
+   *
+   * @param vertx The vertx instance that will be providing requests that this set of authentication
+   *     providers will be handling
+   * @param config The {{@link JsonRpcConfiguration}} that describes this rpc setup
+   * @return Optionally an authentication service. If empty then authentication isn't to be enabled
+   *     on this service
+   */
+  public static Optional<AuthenticationService> create(
+      final Vertx vertx, final JsonRpcConfiguration config) {
+    return create(
+        vertx,
+        config.isAuthenticationEnabled(),
+        config.getAuthenticationCredentialsFile(),
+        config.getAuthenticationPublicKeyFile(),
+        config.getAuthenticationAlgorithm());
+  }
+
+  /**
+   * Creates a ready for use set of authentication providers if authentication is configured to be
+   * on
+   *
+   * @param vertx The vertx instance that will be providing requests that this set of authentication
+   *     providers will be handling
+   * @param config The {{@link JsonRpcConfiguration}} that describes this rpc setup
```

Comment:
s/JsonRpc/WebSocket
s/rpc/

Reply:
Fixed the copypasta on the class link. Didn't change the rpc though, since both of these stacks are running rpc over different transport mediums. JSON-RPC runs over either HTTP or WebSockets here, but we are not consistent about those semantics as we should be.

Reply:
👍